components/StatementsDiscussion/StatementNodeWrapper.tsx

```diff
@@ -52,9 +52,20 @@ const StatementNodeWrapper = ({
   }, [discussion?.comments, comment])
 
   const FocusLink = useMemo(() => {
-    return getLinkComponent(discussion, comment)
+    return getLinkComponent(getFocusHref(discussion, comment))
   }, [discussion, comment])
 
+  const ProfileLink = useMemo(() => {
+    let href
+    // Create a link to the user profile or for anonymous users a link to the comment
+    if (comment?.displayAuthor?.slug) {
+      href = `/~${comment.displayAuthor.slug}`
+    } else {
+      href = getFocusHref(discussion, comment)
+    }
+    return getLinkComponent(href)
+  }, [discussion, comment?.displayAuthor])
```

Comment:
If you want a quick fix do this:
```suggestion
  const profileHref = comment?.displayAuthor?.slug
    ? `/~${comment.displayAuthor.slug}`
    : getFocusHref(discussion, comment)
  const ProfileLink = useMemo(() => {
    return getLinkComponent(profileHref)
  }, [profileHref])
```
Similar change is possible for focus link. Long term I still don't think this is a good solution and I do like the current API interface with `getFocusHref(discussion, comment)`. Let's discuss this when finishing the refactor and spend some time on context and component API design.

Reply:
Fixed it by passing the hrefs and Link element separatly